### PR TITLE
StrictTypesAssember

### PR DESCRIPTION
**Pull request by janvernieuwe**
Build on the work of @icanhazstring in PR https://github.com/phpro/soap-client/pull/250 
Updated to be mergeable with master.
Replaced FileAssembler in favor of StrictTypesAssembler

Dunno why those test are failing, seems unrelated.

### CODE REVIEW

**Comment by icanhazstring** [comment]
Cool. Thanks for picking it up 👍

File: .github/workflows/grumphp.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
                   key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
                   restore-keys: ${{ runner.os }}-composer-
             - name: Install dependencies
-              run: composer update --prefer-dist --no-progress --no-suggest ${{ matrix.composer-options }} --ignore-platform-reqs
```
Comment (janvernieuwe) [inline .github/workflows/grumphp.yml:39]: The ignore platform reqs were screwing up the tests since symfony 6.1 is php 8.1 or higher.

**Comment by janvernieuwe** [comment]
Tests are fixed, was an issue in the test config.

**Comment by veewee** [comment]
Thanks!

**Pull request merged**